### ISSUE TITLE
pg_upgrade does not create citus metadata tables

### ISSUE THREAD

**Issue by mtuncer** [citusdata/citus#964]
Citus metadata tables reside in pg_catalog. These tables are not created on the new instance during pg_upgrade.

Citus metadata tables should be created on the target instance.

following code snipped in PG_Init() should fix table creation part.
```c
if (IsBinaryUpgrade)
{
    SetConfigOption("allow_system_table_mods", "true", PGC_POSTMASTER, PGC_S_OVERRIDE);
} 
```

Check out [postgresql documentation](https://www.postgresql.org/docs/9.6/static/extend-extensions.html) for ```pg_catalog.pg_extension_config_dump()```

see issue [#943](https://github.com/citusdata/citus/issues/943) for more context.

**Comment by mtuncer** [comment]
```pg_extension_config_dump()``` does not copy ```pg_catalog``` tables. 

Limiting this issue's scope to table creation only, and opening a separate issue on copying contents.

**Comment by mtuncer** [comment]
fixed by #973
